packages/client/src/containers/RelatedItems/RelatedItems.Container.js

```diff
@@ -1,33 +1,74 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import ProductCard from '../../components/ProductCard/ProductCard.component';
 import { apiURL } from '../../apiURL';
+import { ViewBackCarousel } from '../../components/ViewCarousel/ViewBackCarousel/ViewBackCarousel';
+import { ViewAfterCarousel } from '../../components/ViewCarousel/ViewForwadCarousel/ViewAfterCarousel';
 import './RelatedItems.styles.css';
 
 export const RelatedItems = ({ category }) => {
   const [categoryProducts, setCategoryProducts] = useState([]);
+  const [offSet, setOffset] = useState(0);
+  const [dataFlag, setDataFlag] = useState(true);
 
-  useEffect(() => {
-    const fetchProductsByCategory = async () => {
-      const response = await fetch(`${apiURL()}/products?category=${category}`);
-      const categoryProductData = await response.json();
+  const fetchProductsByCategory = useCallback(async () => {
+    const response = await fetch(
+      `${apiURL()}/products?category=${category}&limit=3&offset=${offSet}`,
+    );
+    const categoryProductData = await response.json();
+
+    if (categoryProductData.data.length < 3) {
+      fetchAllProducts();
```

Comment:
So we don't show the 2 related items we did find? Perhaps it would be better to show those and then say there isn't any more related items, or if it's 0 say there is no related items but show some other products?

Reply:
Here if there are less than 3 I am showing the first three products from all products and disabling the forward button